ui/tui/cli_handler.go

```diff
@@ -61,10 +61,19 @@ type handlerState struct {
 }
 
 func handleRootVersionFlagAlone(parsedSpec any, _ *CLI) (name string, val any, run func(c *CLI, value any) error, isset bool) {
-	p := parsedSpec.(*FlagSpec)
+	p := AsFlagSpec[FlagSpec](parsedSpec)
+	if p == nil {
+		panic(errors.E(errors.ErrInternal, "please report this as a bug"))
+	}
+
 	if p.VersionFlag {
 		return "--version", p.VersionFlag, func(c *CLI, _ any) error {
-			fmt.Println(c.version)
+			// TODO(snk): We change this later.
+			if c.Product() != "terramate" {
```

Comment:
_nitpick_ I'd say to use a package `const` to avoid magic strings but not a critical thing